building_sim_plugins/building_plugins_common/include/building_sim_common/crowd_simulator_common.hpp

```diff
@@ -0,0 +1,384 @@
+  }
+
+private:
+  double _x, _y, _z, _roll, _pitch, _yaw;
+};
+//================================================================
+/*
+* class MengeHandle
+*/
+class MengeHandle : public std::enable_shared_from_this<MengeHandle>
+{
+public:
+
+  static std::shared_ptr<MengeHandle> init_and_make(
+    const std::string& resourcePath,
+    const std::string& behaviorFile,
+    const std::string& sceneFile,
+    const float simTimeStep
+  );
+
+  MengeHandle(const std::string& resourcePath,
+    const std::string& behaviorFile,
+    const std::string& sceneFile,
+    const float simTimeStep = 0.0
+  )
+  : _resourcePath(resourcePath),
+    _behaviorFile(behaviorFile),
+    _sceneFile(sceneFile),
+    _simTimeStep(simTimeStep),
+    _agentCount(0)
+  {
+    _behaviorFile = this->_ResourceFilePath(_behaviorFile);
+    _sceneFile = this->_ResourceFilePath(_sceneFile);
+  }
+
+  void SetSimTimeStep(float simTimeStep);
+  float GetSimTimeStep() const;
+  size_t GetAgentCount();
+  void SimStep() const; //proceed one-time simulation step in _sim
+  AgentPtr GetAgent(size_t id) const;
+
+private:
+  std::string _resourcePath;
+  std::string _behaviorFile;
+  std::string _sceneFile;
+  float _simTimeStep;
+  size_t _agentCount;
+  std::shared_ptr<Menge::Agents::SimulatorInterface> _sim;
+
+  std::string _ResourceFilePath(const std::string& relativePath) const;
+  bool _LoadSimulation(); //initialize simulatorinterface
+};
+
+//================================================================
+/*
+* class ModelTypeDatabase
+*/
+class ModelTypeDatabase
+{
+public:
+  struct Record
+  {
+    std::string typeName;
+    std::string fileName;
+    AgentPose3d pose;
+    std::string animation;
+    double animationSpeed;
+  };
+
+  using RecordPtr = std::shared_ptr<Record>;
+
+  //Create a new record and returns a reference to the record
+  RecordPtr Emplace(std::string typeName, RecordPtr record_ptr);
+  size_t Size() const;
+  RecordPtr Get(const std::string& typeName) const;
+
+private:
+  std::unordered_map<std::string, RecordPtr> _records;
+};
+
+//================================================================
+/*
+* class CrowdSimInterface
+* provides the relationship between menge agents and gazebo models
+* provides the interface to set position between gazebo models and menge agents
+*/
+class CrowdSimInterface
+{
+public:
+  struct Object
+  {
+    AgentPtr agentPtr;
+    std::string modelName;
+    std::string typeName;
+    bool isExternal = false;
+  };
+  using ObjectPtr = std::shared_ptr<Object>;
+
+  CrowdSimInterface()
+    : _modelTypeDBPtr(std::make_shared<crowd_simulator::ModelTypeDatabase>()),
+    _initialized(false),
+    _sdf_loaded(false)
+  {}
+
+  std::shared_ptr<ModelTypeDatabase> _modelTypeDBPtr;
+  rclcpp::Logger logger() const;
+  void init_ros_node(const rclcpp::Node::SharedPtr node);
+
+  template<typename SdfPtrT>
+  bool readSDF(SdfPtrT& sdf);
+
+  bool initCrowdSim();
+  bool isInitialized() const;
```

Comment:
it looks like this function is not used anywhere, and since `initCrowdSim` returns a `bool` to determine the initialization state, there is no real use for the member `_initialized` as well